mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/entity/naming/SetterStyle.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.entity.naming;
+
+import com.datastax.oss.driver.api.mapper.annotations.PropertyStrategy;
+
+/**
+ * The style of setter that the mapper will look for when introspecting a mutable entity class.
+ *
+ * <p>Note that introspection always starts by looking for getters first (see {@link GetterStyle}).
+ * Once a getter has been found, and if the entity is declared as {@link PropertyStrategy#mutable()
+ * mutable}, the mapper will try to find a matching setter: name inferred as described below,
+ * exactly one argument matching the property type, and the return type does not matter.
+ *
+ * @see PropertyStrategy
+ */
+public enum SetterStyle {
```

Comment:
I deliberately went for a distinct enum, even though the constants are the same right now. This gives us flexibility, for example if we were to implement Scala case class setters at some point in the future.